pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -323,12 +360,20 @@ stdenv.mkDerivation rec {
       mv runfiles.bash.tmp tools/bash/runfiles/runfiles.bash
 
       patchShebangs .
+
+      # bazel reads its system bazelrc in /etc
+      # override this path to a builtin one
+      substituteInPlace \
+        src/main/cpp/option_processor.cc \
+        --replace BAZEL_SYSTEM_BAZELRC_PATH "\"$out/bazelrc\""
     '';
     in lib.optionalString stdenv.hostPlatform.isDarwin darwinPatches
      + genericPatches;
 
   buildInputs = [
     buildJdk
+    python3 # bazel build requires python3. However we still use python2 for most of the other tasks
+            # This will have to be refactored later.
```

Comment:
What are “most other tasks” here? Bazel itself doesn’t need any python2 anymore.

If you still need `python2` for your project, it’s probably better to have a special flag in the derivation, which can switch the bazel build to python2.

Reply:
@Profpatsch switching to `python3` here leads to build failures in a few projects I tried, including `rules_haskell`. Sure they can be fixed (later), but I wanted to keep the reduce the changes in this PR to the bare minimal to quickly fix the bazel in nixpkgs.

Reply:
@Profpatsch but actually, bazel 0.27 deprecates the usage of an implicit python toolchain, so user must now explicitly set their python toolchain.

I can switch to python3 if you want.

Reply:
Switch to python3 done.